src/main/scala/com/target/data_validator/validator/ColumnSumCheck.scala

```diff
@@ -0,0 +1,134 @@
+package com.target.data_validator.validator
+
+import com.target.data_validator.{ValidatorCheckEvent, ValidatorError, VarSubstitution}
+import io.circe._
+import io.circe.generic.semiauto._
+import org.apache.spark.sql.{DataFrame, Row}
+import org.apache.spark.sql.catalyst.analysis.UnresolvedAttribute
+import org.apache.spark.sql.catalyst.expressions.aggregate.Sum
+import org.apache.spark.sql.types._
+
+case class ColumnSumCheck(
+  column: String,
+  minValue: Option[Json] = None,
+  maxValue: Option[Json] = None,
+  inclusive: Option[Json] = None
+) extends ColumnBased(column, Sum(UnresolvedAttribute(column)).toAggregateExpression()) {
+
+  private val minOrMax: Either[String, Unit] = (minValue, maxValue) match {
+    case (None, None) => Left("'minValue' or 'maxValue' or both must be defined")
+    case _ => Right()
+  }
+
+  private val lowerBound: Either[String, Double] = minValue match {
+    case Some(json) =>
+      if (json.isNumber) { Right(json.asNumber.get.toDouble) }
+      else { Left(s"'minValue' defined but type is not a Number, is: ${json.name}") }
+    case None => Right(Double.MinValue)
+  }
+
+  private val upperBound: Either[String, Double] = maxValue match {
+    case Some(json) =>
+      if (json.isNumber) { Right(json.asNumber.get.toDouble) }
+      else { Left(s"'maxValue' defined but type is not a Number, is: ${json.name}") }
+    case None => Right(Double.MaxValue)
+  }
+
+  private val minLessThanMax: Either[String, Unit] = (lowerBound, upperBound) match {
+    case (Right(lower), Right(upper)) if lower >= upper =>
+      Left(s"'minValue': $lower must be less than 'maxValue': $upper")
+    case _ => Right()
+  }
+
+  private val inclusiveBounds: Either[String, Boolean] = inclusive match {
+    case Some(json) =>
+      if (json.isBoolean) { Right(json.asBoolean.get) }
+      else { Left(s"'inclusive' defined but type is not Boolean, is: ${json.name}") }
+    case None => Right(false)
+  }
+
+  override def name: String = "columnSumCheck"
+
+  override def quickCheck(r: Row, count: Long, idx: Int): Boolean = {
+
+    def evaluate(sum: Double): Boolean = {
+      if (inclusiveBounds.right.get) { sum > upperBound.right.get || sum < lowerBound.right.get}
+      else { sum >= upperBound.right.get || sum <= lowerBound.right.get}
+    }
+
+    failed = r.schema(idx).dataType match {
+      case ShortType => evaluate(r.getShort(idx))
+      case IntegerType => evaluate(r.getInt(idx))
+      case LongType => evaluate(r.getLong(idx))
+      case FloatType => evaluate(r.getFloat(idx))
+      case DoubleType => evaluate(r.getDouble(idx))
+      case ut => throw new Exception(s"Unsupported type for $name found in schema: $ut")
+    }
+
+    val bounds = minValue.getOrElse("-Inf") :: maxValue.getOrElse("Inf") :: Nil
```

Comment:
Inf != Double.MaxValue, -Inf != Double.MinValue. This difference could come back to bite us.

Reply:
Sanding off this sharp edge is probably the only thing blocking this PR at this point. My other suggestions can be reserved for a refactor later.

Reply:
I'll just make the else the empty string in both cases.  This is just for messaging.